contracts/utils/ExpressCallHandler.sol

```diff
@@ -16,8 +15,11 @@ contract ExpressCallHandler is IExpressCallHandler {
     uint256 internal constant PREFIX_EXPRESS_RECEIVE_TOKEN = 0x67c7b41c1cb0375e36084c4ec399d005168e83425fa471b9224f6115af865619;
     // uint256(keccak256('prefix-express-give-token-with-data'));
     uint256 internal constant PREFIX_EXPRESS_RECEIVE_TOKEN_WITH_DATA = 0x3e607cc12a253b1d9f677a03d298ad869a90a8ba4bd0fb5739e7d79db7cdeaad;
-    mapping(bytes32 => address) private expressGiveToken;
-    mapping(bytes32 => address) private expressGiveTokenWithData;
+
+    modifier onlyDifferentCaller(address destinationAddress, address expressCaller) {
```

Comment:
why restrict this? express caller might be using the service to send tokens itself, while operating the express relayer. ERC20 transfer should still go through fine

Reply:
Saw the report comment, we should use the original `amount` arg for setting the express caller in storage. We don't need to do the balance difference for the purpose of the event (we did genuinely receive x tokens, after they're transferred to the recipient, any fees charged is not our concern).

And in fact, when the normal call is received, we're using the original amount. So, atm we won't be reading the express storage slot correctly if it was a fee-on-transfer token.